sway-ir/src/parser.rs

```diff
@@ -389,8 +389,8 @@ mod ir_builder {
                 }
 
             rule struct_const() -> IrAstConstValue
-                = "{" _ flds:(field_or_element_const() ++ comma()) "}" _ {
-                    IrAstConstValue::Struct(flds)
+                = "{" _ flds:(field_or_element_const() ++ comma())? "}" _ {
+                    IrAstConstValue::Struct(flds.unwrap_or_default())
```

Comment:
A better way to do this is with the `**` operator.  It's a bit more explicit and doesn't need the `unwrap()`.